app/src/app/app.component.ts

```diff
@@ -34,6 +34,16 @@ export class AppComponent implements AfterViewInit {
         } else {
             this.setTheme("light");
         }
+
+        const routerSubscription = this.router.events.subscribe((e) => {
+            if (e instanceof NavigationEnd && e.url.includes('register')) {
+                const registerSubscription = this.authenticationService.handleRegisterRedirect().subscribe({
+                    next: (res) => { this.checkStatus(); },
+                    complete: () => { registerSubscription.unsubscribe(); }
+                });
+                routerSubscription.unsubscribe();
+            }
+        });
         // check user login
         let authSubscription = this.authenticationService.getSession().subscribe({
             next: (res) => { this.checkStatus(); },
```

Comment:
There is a race condition that happens between the `handleRegisterRedirect()` and the `getSession()` calls when a user is registering. Since the `handleRegisterRedirect()` function sets the current user to undefined, the call to `getSession()` that happens immediately after will throw an error because the current user cannot be found. The `handleRegisterRedirect()` function should retrieve the session itself (described more in another comment below) and the check for user login in this component would only need occur if the user is not registering.